packages/frontend/src/state/lp/hooks.ts

```diff
@@ -177,6 +178,159 @@ export const useCollectFees = () => {
+      const sqrtUpperPrice = new BigNumber(TickMath.getSqrtRatioAtTick(upperTick).toString()).div(x96)
+      const { sqrtPriceX96 } = await getPoolState(squeethPoolContract)
+      const sqrtSqueethPrice = new BigNumber(sqrtPriceX96.toString()).div(x96)
+      const squeethPrice = sqrtSqueethPrice.pow(2)
+
+      // Get previous liquidity amount in ETH
+      const wPowerPerpAmountInLPBeforeInEth = await getQuote(new BigNumber(wPowerPerpAmountInLPBefore), true)
+      const positionEthValue = new BigNumber(wethAmountInLPBefore).plus(new BigNumber(wPowerPerpAmountInLPBeforeInEth))       
+      console.log("positionEthValue", positionEthValue.toString())
+
+      let newAmount0, newAmount1, amountIn, wethAmountInLPAfter, wPowerPerpAmountInLPAfter, tokenIn, tokenOut
+      if (sqrtUpperPrice.lt(sqrtSqueethPrice)) {
+        // All weth position
+        // newLiquidity = positionEthValue/(sqrt(upperPrice) - sqrt(lowerPrice))
+        const liquidity = positionEthValue.div(sqrtUpperPrice.minus(sqrtLowerPrice))
+        // wethAmount = L(sqrt(upperPrice) - sqrt(lowerPrice))
+        wethAmountInLPAfter = liquidity.times(sqrtUpperPrice.minus(sqrtLowerPrice))
+        wPowerPerpAmountInLPAfter = 0
+        amountIn = wPowerPerpAmountInLPBefore
+        tokenIn = oSqueeth
+        tokenOut = weth
+      } else if (sqrtSqueethPrice.lt(sqrtLowerPrice)) {
+        // All squeeth position
+        // newLiquidity = positionEthValue/(squeethPrice/sqrt(lowerPrice) - squeethPrice/sqrt(upperPrice))
+        const liquidity = positionEthValue.div((squeethPrice.div(sqrtLowerPrice)).minus((squeethPrice.div(sqrtUpperPrice))))
+        wethAmountInLPAfter = 0
+        // wPowerPerpAmount = L/sqrt(lowerPrice) - L/sqrt(upperPrice)
+        wPowerPerpAmountInLPAfter = (liquidity.div(sqrtLowerPrice).minus(liquidity.div(sqrtUpperPrice)))
+        amountIn = wethAmountInLPBefore
+        tokenIn = weth
+        tokenOut = oSqueeth
+      } else {
+        // newLiquidity = positionEthValue/(squeethPrice/sqrt(squeethPrice) - squeethPrice/sqrt(upperPrice) + sqrt(squeethPrice) - sqrt(lowerPrice))
+        const liquidity = positionEthValue.div((new BigNumber(squeethPrice).div(sqrtSqueethPrice))
+                                                                          .minus((new BigNumber(squeethPrice).div(sqrtUpperPrice)))
+                                                                          .plus(sqrtSqueethPrice)
+                                                                          .minus(sqrtLowerPrice))
+        // Calculate amounts of each asset to LP
+        // x = L(sqrt(upperPrice) - sqrt(squeethPrice))) / sqrt(squeethPrice) * sqrt(upperPrice)
+        // y = L(sqrt(squeethPrice) - sqrt(lowerPrice))
+        newAmount0 = liquidity.times(sqrtUpperPrice.minus(sqrtSqueethPrice)).div((sqrtSqueethPrice.times(sqrtUpperPrice)))
+        newAmount1 = liquidity.times(sqrtSqueethPrice.minus(sqrtLowerPrice))
```

Comment:
If we use the convention the squeeth price is eth per 1 oSQTH then newAmount0 is in oSQTH and newAmount1 is in weth so we don't need to switch on isWethToken0